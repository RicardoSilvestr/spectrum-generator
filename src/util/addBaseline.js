
/**
  * Add a baseline to the spectrum
  * @param {object} [data] - Your spectrum data in the format {x:[x1, x2, ...], y:[y1, y2, ...]}
  * @param {function} [baselineFct] - Mathematical function producing the baseline you want
  * @return {object} data
  */
export default function addBaseline(data, baselineFct) {
  if (!baselineFct) return data;
  var xs = data.x;
  var ys = data.y;
  for (let i = 0; i < xs.length; i++) {
    ys[i] += baselineFct(xs[i]);
  }
  return data;
}
