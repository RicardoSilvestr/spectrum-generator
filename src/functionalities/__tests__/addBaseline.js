import addBaseline from '../addBaseline';


describe('Test addBaseline', () => {
  var corrected = addBaseline({ x: [1, 2, 3], y: [2, 3, 4] }, (x) => 2 * x);
  expect(corrected).toMatchSnapshot();
});

