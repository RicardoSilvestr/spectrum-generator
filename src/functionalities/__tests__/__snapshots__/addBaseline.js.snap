// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test addBaseline 1`] = `
Object {
  "x": Array [
    1,
    2,
    3,
  ],
  "y": Array [
    4,
    7,
    10,
  ],
}
`;
